File: src/code/javascript/sorting_algorithms/quick_sort.js
const quickSort = (arr) => {
    const n = arr.length

    if (n <= 1) {
        return arr
    }

    const pivot = arr[Math.floor(n / 2)]
    const left = arr.filter(x => x < pivot)
    const right = arr.filter(x => x > pivot)

    return [...quickSort(left), pivot, ...quickSort(right)]
}

File: src/code/ruby/sorting_algorithms/quick_sort.rb
def quick_sort(arr)
  n = arr.size
  return arr if n <= 1

  pivot = arr[n / 2]
  left  = arr.select { |x| x < pivot }
  right = arr.select { |x| x > pivot }

  quick_sort(left) + [pivot] + quick_sort(right)
end
